indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx

```diff
@@ -0,0 +1,401 @@
+                isFetching: false,
+                data: {
+                    bookings: occurrences,
+                    candidates: {},
+                    conflicts: {}
+                },
+                dateRange,
+            },
+        };
+
+        // initial values are constructed using booking object from the props,
+        // thus I need to make sure the Form is not rerendered on every state
+        // change because it would revert all fields to the initial values
+        this.form = (
+            <FinalForm onSubmit={this.updateBooking}
+                       render={this.renderBookingEditForm}
+                       initialValues={this.formInitialValues} />
+        );
+    }
+
+    get formInitialValues() {
+        const {booking: {repetition, startDt, endDt, bookedForUser, createdByUser, bookingReason}} = this.props;
+        const recurrence = getRecurrenceInfo(repetition);
+        const isSingleBooking = recurrence.type === 'single';
+        const dates = {startDate: serializeDate(startDt), endDate: isSingleBooking ? null : serializeDate(endDt)};
+        const timeSlot = {startTime: serializeTime(startDt), endTime: serializeTime(endDt)};
+        const usage = bookedForUser.id === createdByUser.id ? 'myself' : 'someone';
+        const user = {...(usage === 'myself' ? createdByUser : bookedForUser), isGroup: false};
+
+        return {
+            recurrence,
+            dates,
+            timeSlot,
+            usage,
+            user,
+            reason: bookingReason
+        };
+    }
+
+    renderRecurrenceNumber = ({input, onChange, ...fieldProps}) => {
+        return (
+            <ReduxFormField {...fieldProps}
+                            input={input}
+                            onChange={(__, {value}) => {
+                                input.onChange(value);
+                                onChange(value);
+                            }}
+                            type="number"
+                            min="1"
+                            max="99"
+                            step="1" />
+        );
+    };
+
+    renderIntervalDropdown = ({input, onChange, ...fieldProps}) => {
+        const recurrenceOptions = [
+            {text: Translate.string('Weeks'), value: 'week'},
+            {text: Translate.string('Months'), value: 'month'}
+        ];
+
+        return (
+            <ReduxFormField {...fieldProps}
+                            input={input}
+                            as={Select}
+                            options={recurrenceOptions}
+                            onChange={(__, {value}) => {
+                                input.onChange(value);
+                                onChange(value);
+                            }} />
+
+        );
+    };
+
+    renderDateForm = ({input, isSingleBooking, onChange, ...fieldProps}) => {
+        const {value: {startDate, endDate}} = input;
+        const component = isSingleBooking ? SingleDatePicker : DatePeriodField;
```

Comment:
Right now dates earlier that today are disabled and thus past dates are not properly displayed when opening the picker. If we want to allow past dates you'll have to pass an `isOutsideRange` prop to the components as by default dates earlier than today are disabled. I implemented this in the `CalendarRangeDatePicker` and `CalendarSingleDatePicker` components as a `disabledDate` prop (i found it more readable than `isOutsideRange` and we were using this name in other places), in case you want to take a look.

Reply:
Ok, good to know ;)